Lib/test/test_os.py

```diff
@@ -3996,14 +3996,43 @@ def test_oserror_filename(self):
                     self.fail(f"No exception thrown by {func}")
 
 class CPUCountTests(unittest.TestCase):
+    def check_cpu_count(self, cpus):
+        if cpus is None:
+            # None is valid: os.cpu_count() cannot determine
+            # the number of CPUs
+            return
+
+        self.assertIsInstance(cpus, int)
+        self.assertGreater(cpus, 0)
+
     def test_cpu_count(self):
         cpus = os.cpu_count()
-        if cpus is not None:
-            self.assertIsInstance(cpus, int)
-            self.assertGreater(cpus, 0)
-        else:
+        self.check_cpu_count(cpus)
+
+    def test_cpu_count_affinity(self):
```

Comment:
assert that the invariant of the returned affinity value <= the non-affinity return value?

Reply:
Good idea, done.